### DISCUSSION TITLE
How to destroy child components

### DISCUSSION THREAD

**Discussion by Padrezz** [livewire/livewire#4344] [1. Help]
Hello, 

I'm working with MODAL component, which is including COMPONENTS inside self.

MODAL blade looks like `@livewire($template, $params)`, $template is COMPONENT which renders selfs and have own functionality.

MODAL have $buffer of components, which are rendered, one by one, after closing previouse. 

Issue is, that component is not showing new html. After $emit from COMPONENT to MODAL "close" event, MODAL is supposed to clear child and render next component from buffer.  $refresh for some reason doesn't help in my case. 

Any idea how to destroy child component with his HTML ?

**Comment by joshhanley** [comment]
@Padrezz you just need to remove the component out of the rendered html and it will be automatically destroyed.

To do this you would just use a blade `@if` conditional, to conditionally render the child component. Setting the condition to false, would then remove the component from the page.

If you need help with that, please share your blade view where you are including your child components and we can have a look 🙂

Hope this helps!
